### DISCUSSION TITLE
Gain setting and attenuation recommendations for cabled connection?

### DISCUSSION THREAD

**Discussion by pengnist** [srsran/srsRAN_Project#223] [General Help] [answered]
Hi Community,

We have been able to establish connection between srsran gnb and srsue with cabled connection, with 60dB attenuation between TX and RX. However, when trying iperf3 from srsue to core, the maximum MCS at UL was about 18, whereas when using zmq to establish the connection, we can reach MCS=27 at UL. With USRPs and cabled connection, if running for too long, error occurs either at gnb or ue. Please see the screenshots:

An error at UE:
![image](https://github.com/srsran/srsRAN_Project/assets/71417513/e62225a9-644c-4537-8aaf-9d6914661049)

An error at gnb:
![image](https://github.com/srsran/srsRAN_Project/assets/71417513/4a84738c-4052-4721-94ef-0158cd677e04)

I think this problem is related to the gain settings at gnb and ue, as well as the attenuation we added to the cable. I am wondering if you have a recommended setting of gains and physical attenuation that were tested to be stable?

Thanks!

**Comment by dhiaboujebha** [comment]
What kind of USRP are you using for your setup? Can you share your configuration file?

**Reply by pengnist** [reply]
We are using B210. 

The TX and RX gains at gnb are: 
    tx_gain: 75
    rx_gain: 75
I tried the TX gains at srsue in a range from 55 to 75 dB but could not get the optimal MCS.

**Reply by dhiaboujebha** [reply]
are you using a clock? Because in my case the RRC connection is complete, but the PDU Session is not established. Do you know anything about that?

**Reply by pengnist** [reply]
We are using an external 10MHz clock and 1PPS. We set clock and sync to external in the ru_sdr section of the yaml files, as instructed in here: https://docs.srsran.com/projects/project/en/latest/tutorials/source/srsUE/source/index.html.

Did you try using zmq? If connection can be established using zmq, then there should be problems with the radio setting. But I saw more issues that UE is stuck at "Attaching UE...", I don't remember I saw RRC connection is complete but PDU session is not established...

**Reply by pgawlowicz** [reply, marked as answer]
The TX and RX gains has to be tuned for each setup. It might vary depending on the attenuation you put in between two USRPs.

I would recommend to start with low values, and then tune it by running traffic (iperf or ping) between core and UE. 
You increase the gains until you achieve maximize the SNR values for both DL and UL.

You can check the SNR values in console traces of both srsUE and gNB, see example [here](https://docs.srsran.com/projects/project/en/latest/tutorials/source/srsUE/source/index.html#iperf3) 
Note that SNR for UL is reported in gNB as `pusch`
